usr/lib/cca_stdll/cca_specific.c

```diff
@@ -1041,6 +1116,26 @@ static CK_RV cca_get_and_check_mkvps(STDLL_TokData_t *tokdata,
                                      unsigned short domain,
                                      void *private)
 {
+#if defined(_AIX)
+    /* program_invocation_short_name is GNU extension */
+    struct psinfo ps;
+    char program_invocation_short_name[256 + 1] = { NULL, };
+
+    snprintf(program_invocation_short_name, 256, "/proc/%lld/psinfo", getpid());
+    int psfd = open(program_invocation_short_name, O_RDONLY);
+    if (psfd == -1) {
+        fprintf(stderr,
+            "Failed to open procfs to read cmdname: %s\n", strerror(errno));
+        return -1;
+    }
+
+    if (read(psfd, &ps, sizeof(ps)) == -1) {
+        fprintf(stderr, "Failed to populate psinfo: %s\n", strerror(errno));
+        return -1;
+    }
+    close(psfd);
+    strncpy(program_invocation_short_name, ps.pr_fname, PRFNSZ);
+#endif
```

Comment:
Is this block intentionally above the variable declarations? I would like to keep the variable declaration at the top of a block, and code below.